hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/RMWebServices.java

```diff
@@ -618,6 +623,53 @@ public AppInfo getApp(@Context HttpServletRequest hsr,
     return new AppInfo(app, hasAccess(app, hsr), hsr.getScheme() + "://");
   }
 
+  @GET
+  @Path("/apps/{appid}/containers/running")
+  @Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+  public AttemptContainers getRunningContainers(@Context HttpServletRequest hsr,
+                                                @PathParam("appid") String appId){
+    init();
+    if (appId == null || appId.isEmpty()) {
+      throw new BadRequestException("appId, " + appId + ", is empty or null");
+    }
+    ApplicationId id;
```

Comment:
What's happen if appId does not match application id format?

Reply:
Was not nicely managed. I added a test plus a comprehensive error message (see 4 lines below)